io.asgardeo.tomcat.oidc.agent/src/main/java/io/asgardeo/tomcat/oidc/agent/OIDCAgentFilter.java

```diff
@@ -132,6 +144,16 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         }
     }
 
+    private RequestContext getRequestContext(HttpServletRequest request) throws SSOAgentServerException {
+
+        HttpSession session = request.getSession(false);
+
+        if (session != null && session.getAttribute(SSOAgentConstants.REQUEST_CONTEXT) != null) {
+            return (RequestContext) request.getSession(false).getAttribute(SSOAgentConstants.REQUEST_CONTEXT);
+        }
+        throw new SSOAgentServerException("Request context null.");
+    }
+
```

Comment:
since we are already doing this in https://github.com/asgardeo/asgardeo-java-oidc-sdk/blob/c08eedd7a238d738411afa3b918d5cf5f9c1944b/io.asgardeo.java.oidc.sdk/src/main/java/io/asgardeo/java/oidc/sdk/HTTPSessionBasedOIDCProcessor.java#L80, how about including the `redirectURI` in the session context and getting it from the agent filter?

Reply:
We can not directly get the session context from agent. Have to get it from session. So same method.